dl/src/main/scala/com/intel/analytics/bigdl/nn/BiRecurrent.scala

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.nn
+
+import com.intel.analytics.bigdl._
+import com.intel.analytics.bigdl.nn.abstractnn.TensorModule
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.T
+
+import scala.reflect.ClassTag
+
+class BiRecurrent[T : ClassTag] (
+  layer: Recurrent[T], revLayer: Recurrent[T])
+  (implicit ev: TensorNumeric[T]) extends TensorModule[T] {
+
+  val timeDim = 2
+  val inputClone = Tensor[T]()
+
+  val birnn = Sequential[T]()
+    .add(ParallelTable[T]()
+      .add(Sequential[T]()
+          .add(Identity[T]())
```

Comment:
The parallelTable contains two recurrents. One is the unchanged recurrent, the other will add Reverse layer before and after it to process the input data. The Identity layer added here is to balance the structure. It can be removed.